intro_to_R_tidyverse/intro_to_tidyverse.Rmd

```diff
@@ -0,0 +1,173 @@
+---
+title: "Intro to Tidyverse"
+output:   
+  html_notebook: 
+    toc: true
+    toc_float: true
+---
+
+**CCDL 2019**
+
+This Rmd is taking you through the basics of the tidyverse: a collection of packages
+that make R use easier and more convenient.  
+
+There are a lot of tutorials and free books Hadley Wickham and friends have on 
+how to use the [tidyverse packages](https://r4ds.had.co.nz/index.html). 
+[Datacamp](https://www.datacamp.com/) also has a bunch of good tutorials on this. 
+
+## Set Up
+R packages are collections of functions. But your R as it is will not have these
+functions available unless you tell it to.  
+In our docker container, we already have tidyverse installed for you. 
+But if you needed to install it or other pacakges available on CRAN, you 
+do it using the `install.packages` function like this: `install.packages("tidyverse")`
+```{r Load tidyverse}
+library(tidyverse)
+```
+
+"GSE30074.tsv" is a file we will use for practicing the tidyverse.
+This data is from `refine.bio` and is gene expression data from medulloblastoma.
+  
+First let's read in the dataset using the tidyverse package: `readr`.
+Although base R has read-in functions, the `readr` functions are faster and more
+straight forward to use so we are going to use those here. 
+Because the file we are reading in is a .tsv file we will be using the 
+`read_tsv` function. 
+But note that there are analogous functions for `csv` files and other files types.
+```{r}
+# The file we are reading in, relative to our current directory, is in the "data"
+# folder, so we need to put that as the argument. 
+gene.data <- read_tsv("data/GSE30074.tsv")
+
+# Let's also read in the metadata for this gene expression data
+metadata <- read_tsv("data/metadata_clean.tsv")
+```
+
+## dplyr pipes
+A nifty thing with the tidyverse is the pipes: `%>%`
+This handy thing allows you to funnel the result of one expression to the next
```

Comment:
Can we add a simpler example first? Prior to example 1 `without pipes`. Such as:
```R
filter(mtcars, cyl == 6)
mtcars %>% filter(cyl == 6)
```